### ISSUE TITLE
Documentation required

### ISSUE THREAD

**Issue by mrramdhas** [wordpress-mobile/AztecEditor-iOS#1224]
Expecting detailed documentation regarding how to use, what features available etc.

**Comment by netspencer** [comment]
yes! documentation would be very helpful

**Comment by virskor** [comment]
yes! it's bad to guess functions~

**Comment by MadeByDouglas** [comment]
Yes. Looking through the example, its clear this is a capable and powerful library which I would very much like to use. And yet, holy flying fuck you have made this complicated. The `EditorDemoController.swift` has nearly 2000 lines of code!!! Like, clearly you are smart people at Wordpress as you wrote this whole shindig. And yet, did anyone not take a step back, after writing a very cool library, and then think, hmm, maybe our example is a bit too much. Maybe a simpler example that just shows the basics of how to get started, add a simple toolbar, and doesn't require several other dependencies, might serve people better? anyway... Ok. rant over.

**Comment by SergioEstevao** [comment]
> and doesn't require several other dependencies

@MadeByDouglas what dependencies you are referring to?

> Maybe a simpler example that just shows the basics

@MadeByDouglas what do you think should be in the basic demo?

**Comment by MadeByDouglas** [comment]
So, my frustration is really centered around the `EditorDemoController.swift`

I tried to extract just the essence of it, to make a simpler demo of my own, and it became clear everything was spaghetti and all dependent on each other. I ended up just copying the whole file with 1700+ plus lines feeling there must be a better way.

Ok so **dependencies**, the toolbar uses `Gridicons` pod, not horrible, but now without that I have nothing as far as a demo text editor with controls. Would be great to see using SF Symbols or something built in for a basic format bar. 

Next, even if I don't want to do anything with Wordpress, it turns out the attachment VCs and I believe the VC used for the headers all require `WordPressEditor` So now, it turns out to use this as is, I need to import these other libraries.

Furthermore, a refactor of `EditorDemoController` would be very helpful, to help people learn what everything does, what is needed for a basic editor, and what all the possible configurations can be. Also so people could just copy a snippet or so and have a working implementation.

I would **suggest** refactoring the `EditorDemoController` as is, but also making a new demo that just shows the simplest possible implementation just with the library but still gives what I imagine most users want, a featured rich text editor, a formatting toolbar out of the box, and ability to see source html it generates.

**Comment by diegoreymendez** [comment]
Thanks for the feedback.

We're aware that there are a lot of things that could be better both in the documentation and the code.  Aztec is in a place where things work, but require some effort to understand.

Part of the reason why Aztec was released as an open source project is so that external contributions are possible, and it's development is not constrained by the ability of the core team to deliver.

Considering the suggestions, I'd encourage you to consider contributing to the project.  Even small improvements are welcome.

> Ok so dependencies, the toolbar uses Gridicons pod, not horrible, but now without that I have nothing as far as a demo text editor with controls. Would be great to see using SF Symbols or something built in for a basic format bar.

Can you explain a bit more why you want default icons outside of Gridicons?

For context: the reason this pod was added as a dependency was to provide some default icons while still making it easy to remove those resources once the client App provides its own set of icons.

> Next, even if I don't want to do anything with Wordpress, it turns out the attachment VCs and I believe the VC used for the headers all require WordPressEditor So now, it turns out to use this as is, I need to import these other libraries.

Can you provide the name of the specific VCs you need to use that require `WordPressEditor`?

I tried looking into this a bit more but soon realized having more specifics could make the conversation easier.  As a quick example of what I mean: `AttachmentDetailsViewController` does not include `WordPressEditor`.

**Comment by mrramdhas** [comment]
<img width="897" alt="Screenshot 2020-01-04 at 1 56 43 PM" src="https://user-images.githubusercontent.com/47564242/71762805-1e1adc80-2efa-11ea-981b-3a8489e4d0ba.png">

What this code do? Its incomplete. I don't understand what you trying to convey?
